### PR TITLE
Restore the response event.

### PR DESCRIPTION
**Pull request by RubenVerborgh**
In #30, I remarked that the `response` event—and hence, the passed callback—do not behave as in the original Node.js API.

This pull request fixes that, by returning a proxied request rather than the original request. They proxy and the request are identical, except for event handling: the proxy only fires the `response` event when the final redirected response arrives, whereas the (now internal) `response` event of the request fires as soon as the first (non-redirected) response arrives.

### CODE REVIEW

File: create.js [hunk 1]
```diff
@@ -23,10 +23,25 @@ module.exports = function(_nativeProtocols) {
...
+    var clientRequest = cb();
+    var requestProxy = Object.create(clientRequest);
+
+    // separate the proxy's event handlers from the request's
+    requestProxy._events = {};
+    requestProxy._eventsCount = 0;
+    if (options.userCallback) {
+      requestProxy.on('response', options.userCallback);
+    }
+    // forward all request events on the proxy except `response`
```
Comment (RubenVerborgh) [inline create.js]: An alternative would be to forward _no_ events, except `response` and `error` (already performed with `forwardError` below).

Comment (jamestalmage) [inline create.js]: See my response to #30.

I think we should probably only emit events that are logical given that we are making redirection transparent. Also, we would ideally  provide a similar server side experience that `browserify-http`, `browserify-http-2` provide in the browser. I'm guessing many of the events simply aren't detectable in the browser. So I would definitely say just skip it.

Comment (RubenVerborgh) [inline create.js]: I will update the pull request to only provide `response` and `error`.

Comment (jamestalmage) [inline create.js]: > I will update the pull request to only provide response and error.

:+1: Thanks.

Are you willing to dig in on `abort` handling as well?

Comment (RubenVerborgh) [inline create.js]: I could have a look at `abort`, but that probably belongs to a different issue (where we'd discuss desired behavior etc.).

**Comment by RubenVerborgh** [comment]
@olalonde aa18f2905193da9850d266975ae3862734043a40 only provides the `response` and `error` events on the returned request.

File: create.js [hunk 2]
```diff
@@ -125,8 +128,6 @@ module.exports = function(_nativeProtocols) {
       }, options);
     }
     assert.equal(options.protocol, wrappedProtocol, 'protocol mismatch');
-    options.protocol = wrappedProtocol;
```
Comment (RubenVerborgh) [inline create.js]: No need for this assignment, as they are equal anyway.

File: create.js [hunk 3]
```diff
@@ -113,8 +117,7 @@ module.exports = function(_nativeProtocols) {
 
   // returns a safe copy of options (or a parsed url object if options was a string).
   // validates that the supplied callback is a function
-  function parseOptions (options, callback, wrappedProtocol) {
-    assert.equal(typeof callback, 'function', 'callback must be a function');
+  function parseOptions (options, wrappedProtocol) {
```
Comment (RubenVerborgh) [inline create.js]: No need for the callback to be passed to `parseOptions`, as it is not checked anymore.

**Comment by RubenVerborgh** [comment]
@jamestalmage Anything else I can do to complete this pull request?

**Comment by jamestalmage** [comment]
Sorry for the delay.

published as `0.1.0`.

**Pull request merged**